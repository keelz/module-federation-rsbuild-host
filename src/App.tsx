import './App.css';
import RemoteOneButton from 'remote_one/button';
import RemoteTwoButton from 'remote_two/button';
import ButtonUsoage from './ButtonUsage';

const App = () => {
  return (
    <div className="content">
      <h1>Module Federation Rsbuild with React</h1>
      <p>Start building amazing things with Rsbuild.</p>
      <ButtonUsoage />
      <RemoteOneButton />
      <RemoteTwoButton />
    </div>
  );
};

export default App;
