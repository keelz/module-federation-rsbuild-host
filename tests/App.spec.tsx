import { render, screen } from '@testing-library/react';
import App from '../src/App';

// we must mock our federated components as they don't actually exist in this project
jest.mock('remote_one/button', () => 'div', { virtual: true });
jest.mock('remote_two/button', () => 'div', { virtual: true });

test('Renders the main page', () => {
  const testMessage = 'Module Federation Rsbuild with React';
  render(<App />);
  expect(screen.getByText(testMessage)).toMatchSnapshot();
});
