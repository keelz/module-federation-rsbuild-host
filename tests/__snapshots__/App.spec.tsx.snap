// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the main page 1`] = `
<h1>
  Module Federation Rsbuild with React
</h1>
`;
